openapi-maven-plugin/src/main/java/io/github/kbuntrock/yaml/YamlWriter.java

```diff
@@ -71,6 +71,8 @@ public class YamlWriter {
 	public YamlWriter(final MavenProject mavenProject, final ApiConfiguration apiConfiguration) {
 		this.apiConfiguration = apiConfiguration;
 		this.mavenProject = mavenProject;
+		this.om = apiConfiguration.getFileFormat() == FILEFORMAT_JSON ? new ObjectMapper() : new ObjectMapper(new YAMLFactory().enable(YAMLGenerator.Feature.MINIMIZE_QUOTES)
```

Comment:
`==` operator does not permit to check String equivalence in java, only equivalence of the reference.

You need to use "equals" and should replace it by `FILEFORMAT_JSON.equals(apiConfiguration.getFileFormat())` in case of null value.

Reply:
doh, i'm an idiot sometimes :)